### ISSUE TITLE
4.15 creates unreproducible rpm file magic headers

### ISSUE THREAD

**Issue by bmwiedemann** [rpm-software-management/rpm#934]
While working on reproducible builds for openSUSE, I found that 
rpm 4.15 creates rpm headers with mime types in random order.
Results become reproducible when I build on a 1-core VM, so it is probably related to the parallelism added in 4.15.

To see these headers I used `strings` on rpms that were building identical with 4.14.2 .

Diff looks thus: 
```diff
--- filter/RPMS.2017/khmeros-fonts-5.0-0.0.noarch.rpm
+++ filter/RPMS/khmeros-fonts-5.0-0.0.noarch.rpm
@@ -146,13 +146,13 @@
 cpio
 noarch-suse-linux
 directory
-TrueType Font data, 23 tables, 1st "GDEF", 37 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2007 Danh Hong?? ??This font is fr
-TrueType Font data, 18 tables, 1st "GDEF", 35 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2007 Danh Hong?? ??This font is fr
+ASCII text
 TrueType Font data, 18 tables, 1st "GDEF", 35 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2005 Danh Hong??Copyright 2005 Ope
-TrueType Font data, 23 tables, 1st "GDEF", 35 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2007 Danh Hong?? ??This font is fr
 TrueType Font data, 18 tables, 1st "GDEF", 36 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2007 Danh Hong?? ??This font is fr
 TrueType Font data, 18 tables, 1st "GDEF", 35 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2007 Danh Hong????This font is fre
-ASCII text
+TrueType Font data, 18 tables, 1st "GDEF", 35 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2007 Danh Hong?? ??This font is fr
+TrueType Font data, 23 tables, 1st "GDEF", 37 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2007 Danh Hong?? ??This font is fr
+TrueType Font data, 23 tables, 1st "GDEF", 35 names, Macintosh, KhmerOS - font for the Khmer language of Cambodia??Copyright 2007 Danh Hong?? ??This font is fr
   if [ -e /run/fontpackages/reconfigure-fonts ]; then 
     
   if test -x /usr/sbin/fonts-config ; then 
```

**Comment by pmatilai** [comment]
Right, the "dictionary" (RPMTAG_CLASSDICT) used to hold all the file classification strings (for the files we store that data) will get constructed in a non-deterministic order with the parallel file classifying.

This is one of those things... it'd be sad to sacrifice parallel classification because of data that practically nobody uses.

**Comment by bmwiedemann** [comment]
There are ways to do parallel processing and still get reproducible results - it is done in pigz or xz and even make -j .

**Comment by pmatilai** [comment]
Of course. It's more a question of just how much trouble it is to do that - for a data that almost nobody uses.

**Comment by pmatilai** [comment]
Should be fixed by #936.
It's worth noting that the RPMTAG_CLASSDICT data isn't very reproducible in itself as it's highly dependent on libmagic version.

**Comment by bmwiedemann** [comment]
The base level of reproducibility just needs us to be able to use the same inputs (that includes the libmagic binary) to give the same output - like a mathematical function.
On top of that, you can add advanced levels - e.g. try to get noarch packages that are actually the same when built on different archs (is not always true with pngs coming out of image scalers and svg renderers) - and one can find [interesting](https://bugzilla.opensuse.org/show_bug.cgi?id=1109949) [bugs](https://bugzilla.opensuse.org/show_bug.cgi?id=1109541) with it.
